pages/_app.js

```diff
@@ -2,13 +2,19 @@ import GlobalStyle from "../styles";
 import Head from "next/head";
 import { useState, useEffect } from "react";
 import { createContext } from "react";
+import useLocalStorageState from "use-local-storage-state";
 
 export const DataContext = createContext();
 
 export default function App({ Component, pageProps }) {
-  const [movies, setMovies] = useState([]);
-  const [search, setSearch] = useState("");
-  const url = `https://api.themoviedb.org/3/search/movie?api_key=657b6c5e2a2ab2cafa267e54252ca1a7&language=de-GER&query=${search}`;
+  const [movies, setMovies] = useLocalStorageState("newMovies", {
+    defaultValue: [],
+  });
+  const [search, setSearch] = useLocalStorageState("newSearch", {
+    defaultValue: "",
+  });
```

Comment:
Do you really need the search value in the `localStorage`? Because this is so a fast changing value, might not necessary to put it in localstorage.